lib/srv/discovery/common/watcher.go

```diff
@@ -95,18 +96,37 @@ type Watcher struct {
 	ctx context.Context
 	// resourcesC is a channel where fetched resourcess are sent.
 	resourcesC chan (types.ResourcesWithLabels)
+	// preFetchHookFn is called before starting a new poll.
+	preFetchHookFn func()
+}
+
+// Option is a functional option for the Watcher.
+type Option func(*Watcher)
+
+// WithPreFetchHookFn sets a function that gets called before each new iteration.
+func WithPreFetchHookFn(f func()) Option {
```

Comment:
where is `WithPreFetchHookFn` used?

Reply:
In a follow up PR 🙈 
I'll remove this change